### PR TITLE
Fix: Add padding and truncation in huggingface tokenizer

### LINKED ISSUE #3438: ValueError when running HuggingfaceModel with multiple inputs of different length
/kind bug

**What steps did you take and what happened:**
- Modify `test_bert_sequence_classification()` in `huggingfaceserver/test_model.py` such that the two input instances have different length:
```
% git diff huggingfaceserver/test_model.py
--- a/python/huggingfaceserver/huggingfaceserver/test_model.py
+++ b/python/huggingfaceserver/huggingfaceserver/test_model.py
@@ -62,7 +62,8 @@ def test_bert_sequence_classification():
     model.load()
 
     request = "Hello, my dog is cute."
-    response = asyncio.run(model({"instances": [request, request]}, headers={}))
+    request_2 = "Hello, my dog is cute. I love him."
+    response = asyncio.run(model({"instances": [request, request_2]}, headers={}))
     assert response == {"predictions": [1, 1]}
```
- Run the modified test: `pytest -v -W ignore -k "test_bert_sequence_classification"`
- This will fail with:
```
ValueError: Unable to create tensor, you should probably activate truncation and/or padding with 'padding=True' 'truncation=True'
to have batched tensors with the same length. Perhaps your features (`input_ids` in this case) 
have excessive nesting (inputs type `list` where type `int` is expected).
```
Full logs: [test.txt](https://github.com/kserve/kserve/files/14293621/test.txt)

- In short, the problem is that because the inputs have different lengths, they can’t be converted to fixed-size tensors

**What did you expect to happen:**
I expect that we are able to supply inputs of different length in a single batch.

**Potential fix:**
Based on the error, it seems like we need to add paddings, which is documented [here](https://huggingface.co/docs/transformers/en/pad_truncation) by hugging face.

> Batched inputs are often different lengths, so they can’t be converted to fixed-size tensors. Padding and truncation are strategies for dealing with this problem, to create rectangular tensors from batches of varying lengths. Padding adds a special padding token to ensure shorter sequences will have the same length as either the longest sequence in a batch or the maximum length accepted by the model. Truncation works in the other direction by truncating long sequences.
>
> In most cases, padding your batch to the length of the longest sequence and truncating to the maximum length a model can accept works pretty well. However, the API supports more strategies if you need them. The three arguments you need to are: padding, truncation and max_length.

So we should add `padding=True` (or `"longest"`, same behaviour, but more explicit) when calling `self.tokenizer()`, which will pad the inputs to the longest sequence in the batch, so that all inputs will have the same length.

I'm willing to contribute a fix for this. I've tested it locally and it works, but I'll add more rigorous tests as well, perhaps also see if we need truncation as well or just padding.

**Environment:**

- KServe Version: commit [e21b913](https://github.com/kserve/kserve/commit/e21b913fe10b377d72e67fbf92955dbcedd6f465)

### PR DESCRIPTION
**Pull request by kevinmingtarja**
**What this PR does / why we need it**:
Add padding and truncation in huggingface tokenizer. Without this padding, we will run into a `ValueError` when we pass multiple inputs of different length to a `HuggingfaceModel` because they can’t be converted to fixed-size tensors. And without truncation, we will also run into a `RuntimeError` when we pass an input with a token length that is larger than the model's max token length. More details can be found in the issue linked below.

**Which issue(s) this PR fixes**:
Fixes #3438 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

**Feature/Issue validation/testing**:

- [x] Added two tests in `python/huggingfaceserver/huggingfaceserver/test_model.py`

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Review by terrytangyuan — COMMENTED** [review]
LGTM

/cc @yuzisun to double check

File: python/huggingfaceserver/huggingfaceserver/model.py [hunk 1]
```diff
@@ -133,6 +133,8 @@ def preprocess(self, payload: Union[Dict, InferRequest], context: Dict[str, Any]
                 max_length=self.max_length,
                 add_special_tokens=self.add_special_tokens,
                 return_tensors=TensorType.NUMPY,
+                padding=True,
```
Comment (yuzisun) [inline python/huggingfaceserver/huggingfaceserver/model.py:156]: @kevinmingtarja is this always true or we should make them as options

File: python/huggingfaceserver/huggingfaceserver/test_model.py [hunk 2]
```diff
@@ -89,5 +89,33 @@ def test_bloom():
...
+    response = asyncio.run(model({"instances": [request_one, request_two]}, headers={}))
+    assert response == {"predictions": [1, 1]}
+
+
+def test_input_truncation():
+    model = HuggingfaceModel("bert-base-uncased-yelp-polarity",
+                             {"model_id": "textattack/bert-base-uncased-yelp-polarity",
+                              "task": MLTask.sequence_classification.value})
+    model.load()
+
```
Comment (yuzisun) [inline python/huggingfaceserver/huggingfaceserver/test_model.py]: Linting error
blank line contains whitespace

File: python/huggingfaceserver/huggingfaceserver/model.py (hunk 1 above)
Comment (kevinmingtarja) [reply python/huggingfaceserver/huggingfaceserver/model.py:156]: I think it's safe to always set to true, i.e. even if we only provide a single input, no padding will be applied, so we don't need to handle that case separately.

Reference: https://huggingface.co/docs/transformers/en/pad_truncation

File: python/huggingfaceserver/huggingfaceserver/test_model.py (hunk 2 above)
Comment (kevinmingtarja) [reply python/huggingfaceserver/huggingfaceserver/test_model.py]: Fixed!

**Comment by yuzisun** [comment]
thanks @kevinmingtarja !!

**Comment by kevinmingtarja** [comment]
thanks for the review as well!

**Pull request merged**